packages/core/lib/generators/assets_generator.dart

```diff
@@ -641,10 +653,22 @@ class _Statement {
   final bool isDirectory;
   final bool needDartDoc;
 
-  String toDartDocString() => '/// File path: ${posixStyle(filePath)}';
+  String toDartDocString() => '/// File path: $filePath';
 
   String toGetterString() =>
       '$type get $name => ${isConstConstructor ? 'const' : ''} $value;';
 
   String toStaticFieldString() => 'static const $type $name = $value;';
 }
+
+/// Extract metadata from the asset.
+ImageMetadata? _getMetadata(AssetType asset) {
```

Comment:
Could we abstract and gather metadata fetchers based on integrations instead of separated methods?

Reply:
Not all assets are images, so I thought I'd leave it up to each integration to figure out the correct thing to do.